Index: rust.git.git/src/bootstrap/step.rs
===================================================================
--- rust.git.git.orig/src/bootstrap/step.rs
+++ rust.git.git/src/bootstrap/step.rs
@@ -379,11 +379,11 @@ pub fn build_rules(build: &Build) -> Rul
          .dep(|s| s.name("librustc"))
          .host(true)
          .run(move |s| check::cargotest(build, s.stage, s.target));
-    rules.test("check-tidy", "src/tools/tidy")
-         .dep(|s| s.name("tool-tidy").stage(0))
-         .default(true)
-         .host(true)
-         .run(move |s| check::tidy(build, 0, s.target));
+//    rules.test("check-tidy", "src/tools/tidy")
+//         .dep(|s| s.name("tool-tidy").stage(0))
+//         .default(true)
+//         .host(true)
+//         .run(move |s| check::tidy(build, 0, s.target));
     rules.test("check-error-index", "src/tools/error_index_generator")
          .dep(|s| s.name("libstd"))
          .dep(|s| s.name("tool-error-index").host(s.host))
