Index: rust.git/src/etc/rust-lldb
===================================================================
--- rust.git.orig/src/etc/rust-lldb
+++ rust.git/src/etc/rust-lldb
@@ -12,7 +12,7 @@
 # Exit if anything fails
 set -e
 
-LLDB_VERSION=`lldb --version 2>/dev/null | head -1 | cut -d. -f1`
+LLDB_VERSION=`lldb-3.9 --version 2>/dev/null | head -1 | cut -d. -f1`
 
 if [ "$LLDB_VERSION" = "lldb-350" ]
 then
@@ -35,4 +35,4 @@ echo "type summary add --no-value --pyth
 echo "type category enable Rust" >> $TMPFILE
 
 # Call LLDB with the script added to the argument list
-lldb-3.8 --source-before-file="$TMPFILE" "$@"
+lldb-3.9 --source-before-file="$TMPFILE" "$@"
