Bug: https://github.com/rust-lang/rust/issues/52494

Index: rustc/src/test/ui/target-feature-gate.stderr
===================================================================
--- rustc.orig/src/test/ui/target-feature-gate.stderr
+++ rustc/src/test/ui/target-feature-gate.stderr
@@ -1,5 +1,5 @@
 error[E0658]: the target feature `avx512bw` is currently unstable (see issue #44839)
-  --> $DIR/target-feature-gate.rs:27:18
+  --> $DIR/target-feature-gate.rs:31:18
    |
 LL | #[target_feature(enable = "avx512bw")]
    |                  ^^^^^^^^^^^^^^^^^^^
