Bug: https://github.com/rust-lang/rust/issues/50988

--- a/src/stdsimd/coresimd/x86/mod.rs
+++ b/src/stdsimd/coresimd/x86/mod.rs
@@ -73,7 +73,7 @@
     /// let all_bytes_one = _mm_set1_pi8(1);
     /// let two_i32 = _mm_set_pi32(1, 2);
     /// # }
-    /// # if is_x86_feature_detected!("mmx") { unsafe { foo() } }
+    /// # #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] { if is_x86_feature_detected!("mmx") { unsafe { foo() } } }
     /// # }
     /// ```
     pub struct __m64(i64);
@@ -123,7 +123,7 @@
     /// let all_bytes_one = _mm_set1_epi8(1);
     /// let four_i32 = _mm_set_epi32(1, 2, 3, 4);
     /// # }
-    /// # if is_x86_feature_detected!("sse2") { unsafe { foo() } }
+    /// # #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] { if is_x86_feature_detected!("sse2") { unsafe { foo() } } }
     /// # }
     /// ```
     pub struct __m128i(i64, i64);
@@ -166,7 +166,7 @@
     /// let four_ones = _mm_set1_ps(1.0);
     /// let four_floats = _mm_set_ps(1.0, 2.0, 3.0, 4.0);
     /// # }
-    /// # if is_x86_feature_detected!("sse") { unsafe { foo() } }
+    /// # #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] { if is_x86_feature_detected!("sse") { unsafe { foo() } } }
     /// # }
     /// ```
     pub struct __m128(f32, f32, f32, f32);
@@ -209,7 +209,7 @@
     /// let two_ones = _mm_set1_pd(1.0);
     /// let two_floats = _mm_set_pd(1.0, 2.0);
     /// # }
-    /// # if is_x86_feature_detected!("sse") { unsafe { foo() } }
+    /// # #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] { if is_x86_feature_detected!("sse") { unsafe { foo() } } }
     /// # }
     /// ```
     pub struct __m128d(f64, f64);
@@ -256,7 +256,7 @@
     /// let all_bytes_one = _mm256_set1_epi8(1);
     /// let eight_i32 = _mm256_set_epi32(1, 2, 3, 4, 5, 6, 7, 8);
     /// # }
-    /// # if is_x86_feature_detected!("avx") { unsafe { foo() } }
+    /// # #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] { if is_x86_feature_detected!("avx") { unsafe { foo() } } }
     /// # }
     /// ```
     pub struct __m256i(i64, i64, i64, i64);
@@ -299,7 +299,7 @@
     /// let eight_ones = _mm256_set1_ps(1.0);
     /// let eight_floats = _mm256_set_ps(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0);
     /// # }
-    /// # if is_x86_feature_detected!("sse") { unsafe { foo() } }
+    /// # #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] { if is_x86_feature_detected!("sse") { unsafe { foo() } } }
     /// # }
     /// ```
     pub struct __m256(f32, f32, f32, f32, f32, f32, f32, f32);
@@ -342,7 +342,7 @@
     /// let four_ones = _mm256_set1_pd(1.0);
     /// let four_floats = _mm256_set_pd(1.0, 2.0, 3.0, 4.0);
     /// # }
-    /// # if is_x86_feature_detected!("avx") { unsafe { foo() } }
+    /// # #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] { if is_x86_feature_detected!("avx") { unsafe { foo() } } }
     /// # }
     /// ```
     pub struct __m256d(f64, f64, f64, f64);
