Description: Various fixes for Rust on MIPS
Author: Dragan Mladjenovic <dmladjenovic@wavecomp.com>
Bug-Debian: https://bugs.debian.org/881845
Last-Update: 2018-10-24

Index: rustc/src/librustc_codegen_llvm/llvm_util.rs
===================================================================
--- rustc.orig/src/librustc_codegen_llvm/llvm_util.rs
+++ rustc/src/librustc_codegen_llvm/llvm_util.rs
@@ -68,6 +68,9 @@ unsafe fn configure_llvm(sess: &Session)
         // during inlining. Unfortunately these may block other optimizations.
         add("-preserve-alignment-assumptions-during-inlining=false");
 
+    if sess.target.target.arch == "mips" ||
+        sess.target.target.arch == "mips64" { add("-fast-isel=0"); }
+
         for arg in &sess.opts.cg.llvm_args {
             add(&(*arg));
         }
Index: rustc/src/test/ui/asm/asm-out-assign-imm.nll.stderr
===================================================================
--- rustc.orig/src/test/ui/asm/asm-out-assign-imm.nll.stderr
+++ rustc/src/test/ui/asm/asm-out-assign-imm.nll.stderr
@@ -1,5 +1,5 @@
 error[E0384]: cannot assign twice to immutable variable `x`
-  --> $DIR/asm-out-assign-imm.rs:24:34
+  --> $DIR/asm-out-assign-imm.rs:26:34
    |
 LL |     let x: isize;
    |         - help: make this binding mutable: `mut x`
Index: rustc/src/test/ui/asm/asm-out-assign-imm.stderr
===================================================================
--- rustc.orig/src/test/ui/asm/asm-out-assign-imm.stderr
+++ rustc/src/test/ui/asm/asm-out-assign-imm.stderr
@@ -1,5 +1,5 @@
 error[E0384]: cannot assign twice to immutable variable `x`
-  --> $DIR/asm-out-assign-imm.rs:24:34
+  --> $DIR/asm-out-assign-imm.rs:26:34
    |
 LL |     x = 1;
    |     ----- first assignment to `x`
